src/validate.rs

```diff
@@ -92,67 +92,16 @@ fn validate_config(opts: &Opts, fmt: &mut Formatter) -> Result<Config, Option<Co
         return Err(None);
     };
 
-    // Validate configuration files
-    let to_valdiate = paths.len();
-    let mut validated = 0;
-    let mut full_config = Config::empty();
-    for config_path in paths {
-        let file = match File::open(&config_path) {
-            Ok(file) => file,
-            Err(error) => {
-                if let std::io::ErrorKind::NotFound = error.kind() {
-                    fmt.error(format!("File {:?} not found", config_path));
-                } else {
-                    fmt.error(format!(
-                        "Failed opening file {:?} with error {:?}",
-                        config_path, error
-                    ));
-                }
-                continue;
-            }
-        };
-
-        trace!(
-            message = "Parsing config.",
-            path = ?config_path
-        );
-
-        let mut sub_failed = |title: String, errors| {
-            fmt.title(title);
-            fmt.sub_error(errors);
-        };
-
-        let mut config = match Config::load(file) {
-            Ok(config) => config,
-            Err(errors) => {
-                sub_failed(format!("Failed to parse {:?}", config_path), errors);
-                continue;
-            }
-        };
-
-        if let Err(errors) = config.expand_macros() {
-            sub_failed(
-                format!("Failed to expand macros in {:?}", config_path),
-                errors,
-            );
-            continue;
+    match config::load_from_paths(&paths) {
+        Ok(config) => {
+            fmt.success(format!("Loaded {:?}", &paths));
+            Ok(config)
         }
-
-        if let Err(errors) = full_config.append(config) {
-            sub_failed(format!("Failed to merge config {:?}", config_path), errors);
-            continue;
+        Err(errors) => {
+            fmt.title(format!("Failed to load {:?}", paths));
+            fmt.sub_error(errors);
+            Err(None)
```

Comment:
We can now just return `Option<Config>` instead of `Result<Config, Option<Config>>` from this function.